README.md

```diff
@@ -671,6 +672,67 @@ of hook, see the [git-hooks documentation][GHD].
 
 [GHD]: https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
 
+### Adding Existing Line-Aware Commands
+
+Or in other words "low-code error format support."
+
+If you use tools that analyze files and report their findings line-by-line,
+and that Overcommit does not yet support, you may be able to integrate them
+with Overcommit without writing any Ruby code in a similar way as
+[for existing Git hooks](#adding-existing-git-hooks).
+
+These special line-aware command hooks behave and are configured the same way
+as the Git ones, except only file arguments get passed to them.
+Also they must have the `ad_hoc` option, so that, using the command output:
+- differentiating between warnings and errors becomes possible
+- modified lines can be detected and acted upon as defined by
+  the `problem_on_unmodified_line`, `requires_files`, `include` and `exclude`
+  [hook options](#hook-options)
+
+**Warning**: Only the command's standard output stream is considered for now,
+*not* its standard error stream.
+
+To differentiate between warning and error messages,
+the `warning_message_type_pattern` suboption may be specified:
+the `type` field of the `message_pattern` regexp below must then include
+the `warning_message_type_pattern` option's text.
+
+The `message_pattern` suboption specifies the format of the command's messages.
+It is a optional [(Ruby) regexp][RubyRE], which if present must at least define
+a `file` [named capture group][RubyRENCG].
+The only other allowed ones are `line` and `type`, which when specified
+enable detection of modified lines and warnings respectively.
+
+**Note**: The default value for this option is often adequate:
+it generalizes the quasi-standard [GNU/Emacs-style error format][GNUEerrf],
+adding the most frequently used warning syntax to it.
+
+For example:
+
+```yaml
+PreCommit:
+  CustomScript:
+    enabled: true
+    command: './bin/custom-script'
+    ad_hoc:
+      message_pattern: !ruby/regexp /^(?<file>[^:]+):(?<line>[0-9]+):(?<type>[^ ]+)/
+      warning_message_type_pattern: warning
+```
+
+**Tip**: To get the syntax of the regexps right, a Ruby interpreter like `irb`
+can help:
+
+```ruby
+require('yaml'); puts YAML.dump(/MY-REGEXP/)
+```
+
+Then copy the output line text as the YAML option's value, thereby
+omitting the `---` prefix.
+
+[RubyRE]: https://ruby-doc.org/core-2.4.1/Regexp.html
+[RubyRENCG]: https://ruby-doc.org/core-2.4.1/Regexp.html#class-Regexp-label-Capturing
```

Comment:
Let's link to at least Ruby 2.7 docs if possible (Ruby 2.4 is has reached [EOL](https://www.ruby-lang.org/en/news/2020/04/05/support-of-ruby-2-4-has-ended/))
https://ruby-doc.org/core-2.7.3/Regexp.html
https://ruby-doc.org/core-2.7.3/Regexp.html#class-Regexp-label-Capturing

Reply:
Sure, I just wanted to match [overcommit's current prerequisites](https://github.com/sds/overcommit/#requirements). So probably that section should at some point be updated too.

Reply:
So leaving the version as it is and resolving the conversation, unless you object of course.